examples/omni-call/contracts/OmniCall.sol

```diff
@@ -0,0 +1,202 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity 0.8.22;
+
+/// -----------------------------------------------------------------------
+/// Imports
+/// -----------------------------------------------------------------------
+
+//  ==========  External imports    ==========
+
+import { Ownable } from "@openzeppelin/contracts/access/Ownable.sol";
+
+//  ==========  LayerZero imports    ==========
+
+import { OApp, MessagingFee, Origin, OAppReceiver } from "@layerzerolabs/oapp-evm/contracts/oapp/OApp.sol";
+import { MessagingReceipt } from "@layerzerolabs/oapp-evm/contracts/oapp/OAppSender.sol";
+import { OptionsBuilder } from "@layerzerolabs/oapp-evm/contracts/oapp/libs/OptionsBuilder.sol";
+
+//  ==========  Internal imports    ==========
+
+import { OmniCallMsgCodecLib, Call, Transfer } from "./OmniCallMsgCodecLib.sol";
+import { IOmniCall } from "./interfaces/IOmniCall.sol";
+
+/// -----------------------------------------------------------------------
+/// Contract
+/// -----------------------------------------------------------------------
+
+/**
+ * @title Generic omnichain call.
+ * @author LayerZeroLabs (@EWCunha).
+ * @notice Generic contract that handles cross-chain communication
+ * without the need to set-up security stack and messaging options.
+ */
+contract OmniCall is IOmniCall, OApp {
+    /// -----------------------------------------------------------------------
+    /// Libraries
+    /// -----------------------------------------------------------------------
+
+    using OptionsBuilder for bytes;
+
+    /// -----------------------------------------------------------------------
+    /// State variables
+    /// -----------------------------------------------------------------------
+
+    /// @dev For better readability
+    uint8 internal constant ATOMIC_MESSAGE_TYPE = OmniCallMsgCodecLib.CALL_TYPE;
+
+    /// -----------------------------------------------------------------------
+    /// Constructor
+    /// -----------------------------------------------------------------------
+
+    /**
+     * @notice Constructor logic
+     * @param endpoint_: LZ endpoint address;
+     * @param delegate: address to which permissions are delegated.
+     */
+    constructor(address endpoint_, address delegate) OApp(endpoint_, delegate) Ownable(delegate) {}
+
+    /// -----------------------------------------------------------------------
+    /// State-change public/external functions
+    /// -----------------------------------------------------------------------
+
+    /// @inheritdoc IOmniCall
+    function send(
+        uint8 messageType,
+        uint32 dstEid,
+        Call calldata dstCall,
```

Comment:
Just an interesting idea here. If we explore sending an array of Calldata then we can achieve vertical omni-call composibility.